stdlib/public/SDK/Dispatch/Time.swift

```diff
@@ -120,24 +120,35 @@ extension DispatchWallTime {
 // that m2 > 0.
 private func clampedInt64Product(_ m1: Int64, _ m2: Int64) -> Int64 {
 	assert(m2 > 0, "multiplier must be positive")
-	if m1 == 0 { return Int64(0) }
-	if m1 > Int64.max/m2 { return Int64.max }
-	if m1 < Int64.min/m2 { return Int64.min }
-	return m1 * m2;
+	let (result, overflow) = m1.multipliedReportingOverflow(by: m2)
+	guard overflow == .none else {
+		return m1 > 0 ? Int64.max : Int64.min
+	}
+	return result
 }
 
-// Returns m1 * m2, clamped to the range [Int64.min, Int64.max].
-// Because of the way this function is used, we can always assume
-// that m2 > 0.
-private func clampedInt64Product(_ m1: Double, _ m2: Double) -> Int64 {
-	assert(m2 > 0, "multiplier must be positive")
-	let result = m1 * m2
-	if result.isNaN { return Int64.max }
-	if result > Double(Int64.max) { return Int64.max }
-	if result < Double(Int64.min) { return Int64.min }
-	return Int64(result)
+// Returns its argument clamped to the range [Int64.min, Int64.max].
+private func toInt64Clamped(_ value: Double) -> Int64 {
+	if value.isNaN { return Int64.max }
+	if value > Double(Int64.max) { return Int64.max }
+	if value < Double(Int64.min) { return Int64.min }
```

Comment:
Sorry, I realized this check *still* isn't quite correct because Double doesn't have enough precision to represent Int64.max. A terribly stupid intrusive way to do it would be to check against `Double(bitPattern: Double(Int64.max).bitPattern-1)` instead, but that's black magic. @stephentyrone, what's the right way to write this?

Reply:
I agree it's not 100% correct, but it I think it is the closest I can get without "black magic". Maybe a version of this that can use the black magic should be added to the standard library. Anyway, it should be OK as it stands because we don't need to be completely precise - we just need to catch times that are far enough in the future that we should treat them as "forever".

Reply:
We should add a clamping init from Int to Double, but that's outside the scope of this change. As a temporary hack, it suffices to make this `if value >= Double(Int64.max)`; that does the right thing without black magic.

Reply:
@jrose-apple More generally, instead of `Double(bitPattern: x.bitPattern-1)` use `x.nextDown` if you know `x` is positive.

Reply:
Oh, I was looking for `nextDown` and didn't see it. Thanks, Steve.